synapseutils/sync.py

```diff
@@ -184,20 +185,26 @@ def _checkProvenace(item, path):
                     bundle = syn._getFromFile(item)
                     return bundle
                 except SynapseFileNotFoundError:
-                    SynapseProvenanceError(("The provenance record for file: %s is incorrect.\n"
-                                            "Specifically %s is not being uploaded and is not in Synapse."
-                                            % (path, item)))
+                    # TODO absence of a raise here appears to be a bug and yet tests fail if this is raised
```

Comment:
looks like the intention was to raise the exception here. we should fix tests to account for this error being thrown

Reply:
I agree it should be fixed, but will need to look at it further given that the change is to the actual code and not just the tests. I've created an [issue](https://sagebionetworks.jira.com/browse/SYNPY-1069) for it and will address it separately.